### ISSUE TITLE
Retopo the 3D model (or the 3D issues effort)

### ISSUE THREAD

**Issue by thevirtualfer** [Juanvvc/c172p-detailed#446]
There is some 3D issues opened. Long time ago, i has noted a "retopo" on our old TODO file.

Well, the winter is coming. I want very much do that "retopo" right now, dealing ever it is possible, with the opened 3D issues related to the fuselage.

The bad thing about it, is that the model will finish with more vertices ( i don't know how much, but the vertcount will certainly increase).

The main goodness, is that adding more detail to the model will be easer, on the far future releases, when computer power and graphic card power, will increase anyway.

I was attracted before, by the idea of making small changes at a time. But now, i guess it will be very contra productive (at least, for me) because each change will fall in the process of "tweak all textures", "test", "correct", "tweak all textures", "pass".

Then, i guess that making the whole "retopo", these thing will be made in a blocked way. I prefer to correct some things in a row, rather than one thing and test, then other, then other, etc, etc, etc.

My mind will be switching less, if we proceed this way. I am focusing on the "retopo" now. Later i will be only testing and making corrections.

But, of course, the "retopo" will affect texturing, and i am not the only making textures.

@gilbertohasnofb, what do you think ?

There are two ways here.

(1) Ever i made a change, we need stop changing, tweak the textures, test, make possible corrections, and proceed.

(2) I will focus on the "retopo" only. After done, we both talk to choose the best way to cut and unwrap the mesh. Then we change to "texture issues effort".

I prefer the second way, Mainly because, after all, we will need to choose the best way to cut and unwrap the mesh, going through (1) or (2). Obviously that way (1) : 

(a) Will duplicate tasks, because at the end we need to decide how to cut and unwrap the mesh. This will lead us to tweak the textures all again.

(b) Will delay the "retopo", because each change (or small blocks of changes) will stop the process to test, tweak, etc, etc.

(c) Will distract me a lot, because "retopo" and "unwrap" are separated processes and both consumes my brain (heheheheh !).

Waiting you opinion about.

Cheers to all.

**Comment by onox** [comment]
If you want to retopo in order to make the model look better, then go for it! :smile: 

Personally, I don't think a bit higher vertices count will (noticeably) reduce the fps, but we can always make an extra low-poly version (using the existing `.ac` model) before we release 3.8.

**Comment by gilbertohasnofb** [comment]
> The bad thing about it, is that the model will finish with more vertices ( i don't know how much, but the vertcount will certainly increase).

A little bit more should be fine, but keep in mind that this is now the default aircraft in FG and we don't want to make it too heavy. There are plenty of folks out there who need a relatively light plane, so just don't go crazy in the details, particularly in small things like door handles, etc.

> But now, i guess it will be very contra productive (at least, for me) because each change will fall in the process of "tweak all textures", "test", "correct", "tweak all textures", "pass".
> [...]
> But, of course, the "retopo" will affect texturing, and i am not the only making textures.

I agree, but are you sure we will need to do a lot of correction in the textures? As I said before, our paintkit is quite good, and so even if you break the fuselage into little parts you can still UV map it perfectly to our current paintkit. This way we may even get away without redoing the textures.

> There are two ways here

Basically option 2 is better. Try to get the main retopo done and so then I can start tweaking all textures if necessary. But keep in mind what I wrote abote: a good UV map will avoid lots of work. Also note that we have some textures as xcf files, but the older ones are just a single layer in a png file, meaning that changing them is a pain in the neck. Please please please consider this when redoing the UV map.

**Comment by thevirtualfer** [comment]
> I agree, but are you sure we will need to do a lot of correction in the textures?

The number of changes in the textures, is also proportional to the number of textures itself. If we have more textures, we need change more.

> As I said before, our paintkit is quite good.

If you look better, it is not so good. There are a lot of issues related to the UV map. 

When we changed the plane the first time, we tried don't change the livery system a lot. But we never talked about the layout itself. We didn't seat as a team and decided by the better way to unwrap the mesh. We decided by compatibility, not by the better layout.

The textures are really awesome. But we are in the same case when we started, because the front panel of the plane. The layout needed change. The difference was that in that case, this needed was a lot obvious.

Take in mind that, changing shape will lead to change textures.

> Personally, I don't think a bit higher vertices count will (noticeably) reduce the fps...

Nor do i. But i planned to test it on the next weekend. Download flightgear and test the model with a plain white texture. IMHO, the increase of the texture size affects much more the performance that a little increase on the vertcount.

**Comment by gilbertohasnofb** [comment]
> If you look better, it is not so good. There are a lot of issues related to the UV map. 

Where? Which issues? The issues I see are much more related to not perfect UV mapping than to the textures themselves. If you look we have several opened issues about better mapping some parts of the plane using the same textures.

> The textures are really awesome. But we are in the same case when we started, because the front panel of the plane.

But how is that going to change the paint kit?? Can't we just use the same ones we have? Surely you can UV map the new 3D objects using what we have, right?

> IMHO, the increase of the texture size affects much more the performance that a little increase on the vertcount.

We offer both HD and non-HD textures, and you can change that during a flight. Offering two models on the other hand is a more delicate issue in my view. So yeah, I think we should keep the number of vertices as low as possible and not go crazy with little things. But that's just my 2 cents.

**Comment by Juanvvc** [comment]
I think we must reuse the existing paint kit as much as possible, even if the model changes. @thevirtualfer take into account redoing all textures is a hell of a work! Try to uv-map to the same images and only change the map if it is really necessary.

In any case, I strongly believe we will need a new uv-map for the front of the aircraft. We are just extending artificially the map to cover the bow of the aircraft. After the new details added to the front model, I believe this trick is not going to look very well any more.

**Comment by gilbertohasnofb** [comment]
>  I guess a new front view must be added to the paint kit.

That's easy enough and I am willing to do that for all liveries.

**Comment by gilbertohasnofb** [comment]
@thevirtualfer also, if you are going to redo the fuselage around the nose, could you please have a look at this issue here: https://github.com/Juanvvc/c172p-detailed/issues/6

It is for me the absolutely worst problem we have with the 3D model, it makes the cockpit view look so bad and so different than in real life...

**Comment by thevirtualfer** [comment]
> Where? Which issues? The issues I see are much more related to not perfect UV mapping than to the textures themselves.

Exactly what i am saying here. The UV mapping is not good. And doing a better unwrap will affect the textures.

> Surely you can UV map the new 3D objects using what we have, right?

As much as possible. But no way to use it exactly as we have.

> I think we must reuse the existing paint kit as much as possible, even if the model changes.

Again, it is exactly what i am saying. But being realist, you know that we will need change it. I am just preparing to the future. The process of unwrapping is not so near. But it will come and will be a headache to all of us. I am just trying to be sincere about this.

> In any case, I strongly believe we will need a new uv-map for the front of the aircraft.

This is another example of as bad our unwrap is. That is all related.

> take into account redoing all textures is a hell of a work and they are currently very good.

I am not saying the textures are bad or good. The paintkit is : 

(1) The UV-mapping layout
(2) The textures

The number (1) needs be rethinked. I know it may be frustating. As i said, we can try unwrap the mesh to almost fit into current layout. But honestly, i know that we will need change ALL the textures.

**Comment by gilbertohasnofb** [comment]
oh boy.....

**Comment by thevirtualfer** [comment]
Gil.

I will help. Have you a account of the number of liveries ? Ignore the duplicated ones, because it is a matter of scale.

We can assign a half-to-half. I swear that i will do all the possible to not overcharge you.

**Comment by thevirtualfer** [comment]
Another way to deal with this is by making the same decision as before.

Create a new aircraft. A more detailed c172p.

**Comment by thevirtualfer** [comment]
> But keep in mind what I wrote abote: a good UV map will avoid lots of work.

This is exactly what we don't have. It is all about i am saying. We need a better layout to avoid lots of work in the future. The layout must cover all sides of the plane (including bottom) and a special care is needed by junction points or curved surfaces.

**Comment by gilbertohasnofb** [comment]
> Create a new aircraft. A more detailed c172p.

I am not sure I am up for that. Our model isn't bad by any means, and I'd much rather continue with it as it is. Also, if we have energy to create a new model then I propose that we do another plane (FlightGear is begging for a Cessna 182, a Piper Tomahawk or proper Cessna 152). But concerning this project here, given all the work we all invested I just think it would be crazy to redo the whole thing to be honest.

**Comment by onox** [comment]
Just do the retopo and UV stuff in this project (in a feature branch). Near the end of 3.8 we can look at the performance and decide if we want to replace the `.ac` in Models/ or postpone it to 3.10.

**Comment by thevirtualfer** [comment]
It was only an option to put on the table.

But there are two points here.

(1) Our model isn't bad. But it is concerned to the personal opinion. Someone can say that our model is bad in the X or Y aspect. 

(2) There will be not much difference on the amount o work using another model, or applying changes directly over this one. As before, we didn't make the plane from the ground. We got what was already made and added more things.

Basically, the work will be the same, doing the changes over this model, or using another model.

I give the suggestion, only because someone may be apprehensive by changing the default plane.

It is not my case.

**Comment by onox** [comment]
I don't really understand the issue to be honest. @thevirtualfer just makes the model prettier by doing retopo + UV stuff + unwrapping + other cool Blender tricks :smile: Updates `.blend` and `.ac` files in a separate branch. After that we can decide if we just merge it or rename some of the files and have two `-set.xml` files (one pointing to the old `.ac` file and the other pointing to the newer/better one) Am I missing something?

**Comment by thevirtualfer** [comment]
You are right, @onox.

This is not the time to make this type of decision. I will proceed with the retopo. When finished, will talk about UV mapping.

**Comment by thevirtualfer** [comment]
> Am I missing something?

HAHAHHAHA ! Yes, you are.

It is very very probable (in fact, it will occurs) that the whole texture files will need changes. This will be a nightmare.

**Comment by gilbertohasnofb** [comment]
> It is very very probable (in fact, it will occur) that the whole texture files will need changes. This will be a nightmare. 

No, you told me that it's very probably that small modifications will happen, no that the "whole texture files will need changed". If a complete new paintkit will be created and the textures must be redone from scratch then I won't make any more textures for the plane, I can't just simply flush 6 months of work because you want to modify the 3D model. Because if I do that, then what will happen in 6 months from now? Will someone else come up with still a better "solution" still and recreate once again the 3D model and therefore flush once again my work? Sorry but that's a big no no to me, my time is really valuable to me.

**Comment by thevirtualfer** [comment]
Gil. You are upset. I understand this.

I said i will help with textures, didn't i ?

But ok. Lets look to the cenario.

(1) Or we don't change the plane never more, because the paintkit can't be touched.

(2) Or we change the plane and tweak textures as needed.

If you want block plane modifications, it is ok to me. 

Really, i understand your frustration. But plane mods will occur now or later.

Maybe the texture modifications will be not so hard to do. I don't know yet. Lets reach that point first. Then we talk about the UV mapping. I will not do anything without your "agreed".

**Comment by thevirtualfer** [comment]
> If a complete new paintkit will be created and the textures must be redone from scratch 

As i said, it isn't the case. I will try unwrap the mesh to best fit over what you have. But i know that it will not fit completely. We both can decide by the better cut.

> No, you told me that it's very probably that small modifications will happen.

Not exactly what i said, but ok. The amount of modifications on the textures will propagate by the number of texture files we have. Maybe the problem is my english. But what i am saying is : If ONE texture file needs be modified, then all the others will need the same modification.

**Comment by Juanvvc** [comment]
I can understand both the hesitation of @gilbertohasnofb to change (again!) the textures and the interest of @thevirtualfer of providing the best possible model. At the time being, I'm still confident that the improved model can be uv-mapped to the current textures quite well, except probably the front view.

@thevirtualfer, I'm going to the flying club next Saturday. Do you need any specific photo of the exterior of a c172? I think there are a couple of them in the hangars.

(I'm currently flying a Diamond DA40 and a Piper PA28 ;) )

**Comment by thevirtualfer** [comment]
> I'm still confident that the improved model can be uv-mapped to the current textures quite well

God hear your words ! But i am very pessimist about it. But of course, we will try to fit the maximum that we can.

> Do you need any specific photo of the exterior of a c172?

Sure ? Great !

I am in trouble about the top and bottom view of the engine cowling. And most urgently, how the rudder connects with the vstab and the body. Googled a lot about this, without success.

In the bottom, it is basically the point the "NosewheelStruct" connects to the fuselage. There is a hole there ? What is the shape ?

**Comment by gilbertohasnofb** [comment]
I understand it all and we must see how it goes, but I made it clear that I don't think I will have time nor energy to recreate those same liveries from scratch due to a new paint kit. If we need to adapt the older ones, then of course I am in, but else it may be problematic situation. Anyway, let's see how these things go once the 3D model is fixed.

@thevirtualfer I do appreciate your concern with what I think and your offer of help. As I wrote, let's see how it goes once you have the model done.

**Comment by thevirtualfer** [comment]
Ok. I will proceed with the retopo. The fuselage increased the vertcount but doors, glasses, hstab, vtab and rudder, decreased. The steps are : 

(1) retopo (doing good faceloops, using only quads, avoid poles, etc, etc) ;

(2) After done, clean the mesh ;

(3) The problematic cut ;

I will make an account of the vertcount only after step (2).

But only to illustrate, look the door glass, for example.

Before retopo : 

<a href='http://postimage.org/' target='_blank'><img src='http://s13.postimg.org/64hukqm3r/retopo_1.jpg' border='0' alt="retopo 1" /></a><br />

Can one really understand that ?

After the retopo : 

<a href='http://postimage.org/' target='_blank'><img src='http://s13.postimg.org/4dytj94kn/retopo_2.jpg' border='0' alt="retopo 2" /></a><br />

A much more clean and easy to manage.

**Comment by tigert** [comment]
While you retopo, would it be good to put these as reference background images in the .blend?

http://drawingdatabase.com/wp-content/uploads/2015/04/cessna_172_3v.jpg

The license is unclear, but this is very likely originally from the pilots operating handbook from Cessna.

I am sorry I am too new to blender at this point to help much more, but tried placing the blueprint as background image for reference, helps us to spot issues like the too low glareshield.

https://www.youtube.com/watch?v=-TQ-dBihGU0

I tried to do this myself but the model is a bit tilted compared to this one, which is fine for side views, but this distorts the top and front views as compared to the blueprint, thus I think we'd need to rotate the model a bit in the side view (I think the tail is a bit higher in the above three-way drawing)

Is it a lot of work to align the 3d model with this image so that we could start tweaking the .blend a bit here and there? 

(and this is completely new area to me, but it would be fun to learn, though poking at the 3d model likely requires a lot of coordination so that we don't mess each others work up)

**Comment by thevirtualfer** [comment]
Hi, @tigert !

I already tried with this blueprint, but our plane doesn't fit on.

Using a background image as reference is not the best approach (to me) because it shows only in the predefined views (top, left, camera, etc) and scaling and aligning it is a bit boring (to me).

I use "empties" to hold the blueprint. It gives more freedom in the alignment. Look : 

<a href='http://postimage.org/' target='_blank'><img src='http://s14.postimg.org/hsrmdalr5/Screenshot_from_2015_08_26_10_20_14.png' border='0' alt="Screenshot from 2015 08 26 10 20 14" /></a><br />

To achieve this, first add an empty

<a href='http://postimage.org/' target='_blank'><img src='http://s14.postimg.org/kpendkrkx/Screenshot_from_2015_08_26_10_20_44.png' border='0' alt="Screenshot from 2015 08 26 10 20 44" /></a><br />

Then, with the empty selected, go to here and set the image you want

<a href='http://postimage.org/' target='_blank'><img src='http://s14.postimg.org/52ne07dsx/Screenshot_from_2015_08_26_10_21_24.png' border='0' alt="Screenshot from 2015 08 26 10 21 24" /></a><br />

Now, duplicate the empty and rotate as needed.


The distance of the empties are not important, if you uses orthographic view.

You can put the empties on a separated layer that you can hide when they are disturbing what you are doing.

As i said, this blueprint is not in the same proportion of our cessna. Good blender to you, guy.

**Comment by thevirtualfer** [comment]
Guys. I on this area of the plane, but as a rookie, i am in trouble with differences on reference photos.

What one of these is the shape of this part of the vertical stab for the 172p ? Or none ? Or any ?

<a href='http://postimage.org/' target='_blank'><img src='http://s21.postimg.org/wxj9jknnr/vstab_01.png' border='0' alt="vstab 01" /></a><br />

Cheers to all, and thanks in advance.

**Comment by thevirtualfer** [comment]
Tuomas.

I started this effort trying to align the plane with blueprints. I have 7, but 4 are copies from the others. The remaining 3 different blueprints, don't match itself. For example, blueprint A doesn't match witch blueprints B and C, and so on.  And none of the blueprints matches our plane.

But our plane almost match the fsx plane. The diffs are minimal.

I don't know how to proceed. But assumed that the proportions of our plane are correct.

If someone have a different opinion, let me know. The general format can be modified. It can be easy or difficult. I can't decide because i don't know the exact proportions. If someone with more confidence says : 

 - Use this reference as it is !

It will be no problem to me.

Cheers, bro.

**Comment by thevirtualfer** [comment]
@tigert.

I figured out why the blueprint doesn't fit. At least, i have a strong clue.

- The blueprint is so bad, that it contradicts itself.

We can check it, by drawing a red line that covers the tail in front view. See as it flows below the windows and finish in the spinner.

<a href='http://postimage.org/' target='_blank'><img src='http://s14.postimg.org/i7b6fjidd/retopothread_01.png' border='0' alt="retopothread 01" /></a><br />

Now, the same red line in front view, shows that the blueprint is wrong with itself.

<a href='http://postimage.org/' target='_blank'><img src='http://s14.postimg.org/5tyc8msox/retopothread_02.png' border='0' alt="retopothread 02" /></a><br />

The red line is in the level with the windows and above the spinner.

The problem is that the author used some perspective rule that we will not guess.

Ideally, a blueprint needs to be orthographic. Without perspective involved.

Case closed, to this one.

**Comment by tigert** [comment]
The side view has a reference line from spinner to tail. Maybe try rotating that view so the line is horizontal? Does it still not line up?

For tails, rounded1 is likely ours, 2 is close. Might be late models. Number 3 is very early 172 I think.

**Comment by thevirtualfer** [comment]
The tail aligns using this approach.

The wheels, not.

I don't know if this blueprint gives the correct proportion or not.

Our plane is completely different.

But, for example, it should not be possible see under the window in the side view. At least, from that bar to the fuselage. Appears that some perspective was used someway.

BUT...

If dimensions in the blueprint are right, our plane is really wrong.

The major problem is :  I really don't know.

**Comment by thevirtualfer** [comment]
About the trim, don't know if we leave the object attached to the elevator, or make it a new object and do some xml magic you are so good to do.

![garage_report_17](https://cloud.githubusercontent.com/assets/10703842/9563682/c1a74208-4e62-11e5-89f9-75073fa6a27e.jpg)

**Comment by thevirtualfer** [comment]
@tigert 

Working on a separated version, where i will try to match this blueprint with our plane.

Appears that the plane is more beautiful now.
Will adjust the front and bottom of the plane as soon i can, and post here to your judgement.

Cheers.

**Comment by tigert** [comment]
Too bad a very detailed bluepront seems difficult to find, this seemed the best around.

I can make measurements from a real plane if those would help? Luckily there are a lot of photos of c172's around to verify the model, too bad they are all in perspective of course, so it is more about the feeling than exact measurements.

But let me know if you want more detailed pics or measurements. Or maybe closeup textures etc..

And thanks for the efforts everyone, this is fun stuff to follow :)

**Comment by thevirtualfer** [comment]
@tigert 

At the end, there is no way to align both, the side view and the front view.

When i aligned my view to match the elevator in the front view, the bottom of the plane in the blueprint goes very down.

I am using only the side view and top view.

> But let me know if you want more detailed pics or measurements. Or maybe closeup textures etc..

I am just making adjustments for now. but wondering how things are from top view. The vertical stabilizer, for example. There are a lot of photos from side, back, etc. But almost none from top.

I got 2 of then.

About textures, yes, we will need. But not exactly now.

About closeups, wow. I will need then. Will talk with you when necessary.

Thanks to you, bro.

**Comment by thevirtualfer** [comment]
There is other idea here.

You can take a photo from a specific and well know position and distance from the plane.

Then you pass me : 

(1) The distance you are from the plane ;
(2) The line from the plane you are (i.e. facing directly the baggage door, etc)
(3) The height you are when taked the picture
(4) The parameters of your cam

(1) (2) and (3) will give me an idea the x, y, z i can position my blender cam.

(4) Will help me to setup my blender cam

Maybe we get a perfect fit, in this way.

What you think about ? Can you test it ?

**Comment by dany93** [comment]
@thevirtualfer, 
- On your blueprints above, the side and front views are obviously not at the same scale. Maybe, scaling one of them to match the other one (height) could help.
 @tigert too,
- About photos: taking them from a very large distance, with zoom, would minimize the perspective effects (to get a nearly ortho view).

**Comment by tigert** [comment]
How about defining a few key points. Spinner tip, navi lights, landing gear hubs, beacon, etc. Easily pinpointable from photos. Take exact cross-measurements between those points.

Could these be used to align the camera when you mark them in the photo as well?

**Comment by thevirtualfer** [comment]
Still here, doing some tweaks. 
![garage_report_18](https://cloud.githubusercontent.com/assets/10703842/9593815/cb44251a-5029-11e5-910e-b26a9122adc7.jpg)

**Comment by thevirtualfer** [comment]
@dany93 

The front view has a bit more problems. It is not a true "front". The horizontal stabilizer and wings can be aligned, at the cost of not aligning the wing support (at the point it connects with fuselage).

The entire bottom of the plane goes down. I mean, if i aligned the fuselage in side view, i cant align the same fuselage on the front view. The top of the plane is ok. The bottom is not of the same proportion in orthogonal view.

@tigert 

Sorry, bro !
I am not sure i understood what you are proposing. You mean base the model by proportions between some key points distance ?

Maybe it works. Lets try.

If your cam has a good zoom, maybe the @dany93  approach works too.

Cheers, guys.

**Comment by tigert** [comment]
I am not sure if this works, due to lens distortion or some other reason I cannot think of right now (since I have never tried this idea in practice)

But perhaps it does? I am not sure.

Think of definining a few easily identifiable points in the C172: spinner tip, beacon light on top of the rudder, navigation lights on wings, landing gear wheel hubs, tail hook for example. Measure cross-distances of each: from spinner to left and right navigational light etc.

Draw a "stick airplane": One "stick" from spinner tip to tailhook. Another from left to right navigational light. Position these as correctly as possible relative to each other, so you get a super-simplistic mesh "skeleton" for the aircraft.

Now, if you take any photograph of a C172 from an arbitary angle, would it be possible to align the 3d model "skeleton" so that those defined points match the photo? If that is possible, you could use any photo as a reference even if it is not fully ortographic (as nothing from a regular camera really is)

Maybe this is a flawed idea, I am not sure. Do you understand what I am trying to explain? :-)

**Comment by thevirtualfer** [comment]
Maybe it works, at all.

Lets try it, @tigert.

Well, just to explain.

(1) There is no (at least, until now) a good blueprint of the model.
(2) The details, are not a issue. There are lots of closeups (@tigert has a bunch of photos)
(3) The main problem is the proportion of the model, the big bends, the overall format.

Trying to resolve this is turning a "we will try the best we can" saga.

In fact, the model already has some "feeling". But if we can do it better, well, we will do.

I decided to adapt my model to the 3.2 version, because none above runs on my computer. It is a pity. Later, when i figure out the problem, i will port to last version.

**Comment by thevirtualfer** [comment]
I got a terrible flu, so i lower the velocity of changes. I'm on the wings, and now is not only the textures that may be changed, but probably the flap xml code needs change too.

From some photos and videos, i guess the wings is longer in the upper section.

![image](https://cloud.githubusercontent.com/assets/10703842/9695888/43af923a-533d-11e5-8d31-b5c436a9bd19.png)

The flaps is if fact below the wings.

![image](https://cloud.githubusercontent.com/assets/10703842/9695908/7d6d6a38-533d-11e5-8a11-3bee0e5bd5dd.png)

And when it rotates, appear to me it goes down a bit. So, if someone has the correct info, it is ok.

![image](https://cloud.githubusercontent.com/assets/10703842/9696006/4977d43c-533f-11e5-90c1-11f8996bdc75.png)

And as wing was remade, it makes a more opened curve on the front. To me, the old wing was more sharpened than the normal. Also, if someone can clarify this, it is ok, too.

![image](https://cloud.githubusercontent.com/assets/10703842/9695993/ff02b926-533e-11e5-840e-82b5f288fe4d.png)

**Comment by thevirtualfer** [comment]
Again, if someone has photos for this area from the bottom of the wing, it will be welcome.

Cheers, guys.

**Comment by tigert** [comment]
That is a fowler flap, so it both extends down and backwards (extends wing area).

This thread concerns a Cessna 170, but it might also be a  C172 - but nevertheless I think the wing is sort of the same - the structure is similar anyway, even if the measurements might not.
http://www.cessna170.org/forums/viewtopic.php?f=2&t=4659

The flap has "rollers" that move along those grooves when the flap extends. Think of them as two attachment points and movement paths for the animation.

I can try to look for (and take new!) photos and measurements of the flaps - are there some particular things you are interested in? I'll likely be at the airport on wednesday.

**Comment by thevirtualfer** [comment]
Hi, @tigert 

In fact, this flu is killing me. Thanks, bro. If i feel better, will see what i can do about this.

**Comment by legoboyvdlp** [comment]
I certainly hope that is not literally true ;) Your changes sound great. Hope you recover!

**Comment by thevirtualfer** [comment]
Feeling better.
Someone can explain this picture ? Or have other picture like this ?
![image](https://cloud.githubusercontent.com/assets/10703842/9823156/f570ff10-589a-11e5-85be-9d983a792f52.png)

I see a disk, the white plate and the black box. There is something else, here ?

**Comment by thevirtualfer** [comment]
Slow, i agree. But going forth.

Today i made theses two mechanisms. The red one, to control the rudder movement, and this other, to control the trim tab. We are looking the bottom back of the plane.

![garage_report_19](https://cloud.githubusercontent.com/assets/10703842/9828766/c9ee0a30-58c8-11e5-9229-707f67232041.jpg)


Will go to wheel struct tomorrow. As i understood, the photo i posted above, is the left wheel. Will put what i am seeing. The black box, the white plate and the corroded disc. And of course, the reduction on the wheel struct.

**Comment by dany93** [comment]
Hi @thevirtualfer 
I hope you recover soon.

Just some hypothesis (to be confirmed by @tigert )
- The corroded disc: the brake disc,
- The black box:  brake mechanism (hydraulic pressure, jaws),
- The white plate with holes: fairing support?

**Comment by tigert** [comment]
Here is another view: 

http://tigert.1g.fi/kuvat/MIK/CAU/DSC_0667.jpg

The rusty disc is the brake disc, the thick metal part is the brake cylinder, just like in cars, that clamp around the brake disk on the front side of the wheel. The thinner plate with the holes is where the wheel pants get attached to. 

The photo above has a higher res version when you click it, you can see somewhat what's there.

I was at the airport on wednesday and we had a lcd backlight blow up on our sim project, which resulted in a soldering adventure - and I forgot to take the measurements, sorry :-/

Nice progress on the details, I can take more closeup pics with a ruler for measurements if you want, also let me know of any sizes etc you want to cross-check, when I'm at EFHF I can just walk to a Skyhawk with a ruler and check and snap some closeups from any angle you need.

**Comment by tigert** [comment]
This could also be useful, sorry I cannot search the correct page (on phone browser now) but found this via google:

https://www.redskyventures.org/doc/cessna-maintenance-manuals/Cessna_172-175_pre1963_PartsManual.pdf

**Comment by thevirtualfer** [comment]
@tigert. The photo and the explanation was sufficient.

Done these stuff and it is pretty cool. I lost yesterday, trying making 3.7 work. It is running now.

Now i am facing some bugs, with the model itself. Sometimes, when i change the view, it causes a segmentation fault. The colors in the interior flick according the angle i am looking.

Now, i am putting object by object, one by one, and starting fgfs to identify what is wrong.

@Juanvvc, have you any clue ?

**Comment by thevirtualfer** [comment]
Guys.
I am inside the plane now.
(a) Remade the cabin
(b) Added seat belts, sunshades, etc.
(c) Fixed the #418 
(d) Working on #6

Need some help to animate the sunshades.

About the panel, it is a boring job, but i am doing it.
The fact is that the blueprint draws the panel almost horizontal. But photos i saw, shows that the panel is a bit inclined to up. following the photos.

Tomorrow i can post some pics. Now, i am going to a party.

Cheers to all.

**Comment by wlbragg** [comment]
>Need some help to animate the sunshades.

@thevirtualfer. what do you need?

**Comment by thevirtualfer** [comment]
Hi @wlbragg !

Well. I modeled the sunshades and, in fact, can animate it. But i don't know how to set it clickable, hold its status, etc.

I put the animation code, related to the door handles. In this video, when i click the exterior door handle, i am animating the sunshade. 

https://youtu.be/P9YmBPmH9Qs

What i want is click on the sunshade and animate it.

**Comment by thevirtualfer** [comment]
Well, well, well, well.

Rethinking about the interior, trying to fix issue #6, i found this blueprint

![panel1](https://cloud.githubusercontent.com/assets/10703842/10005433/a161981c-608d-11e5-9a95-9cb523c75509.png)

It is the panel on the 1981 model. The holes are, well, just the holes the instruments will fill.

The main problem is that using this blueprint as reference, our current panel is very big.

I am modeling a new panel using these dimensions, and as i moved the panel up, because issue #6, there will be a big space between the panel and the floor.

Don't know what to do.

As soon i finish the new panel, i will post some pics to share the final decision about this "issue". As you know, i don't have a cessna on my garage. Only an old bike, because my car is with my sister.

**Comment by gilbertohasnofb** [comment]
@thevirtualfer I think @tigert has some pictures of the panel together with a ruler for reference, so you can tell the exact size of anything there.

**Comment by wlbragg** [comment]
>But i don't know how to set it clickable, hold its status, etc.
We can get that done for you easy enough. It would help if I had the code to look at. Even the object and or associated property names along with the .xml your using to animate it now, would help.

**Comment by thevirtualfer** [comment]
Hi, @wlbragg 
I put this code on the Model/c172p.xml
`

    <animation>
        <type>rotate</type>
        <object-name>sunshade_left</object-name>
        <property alias="../../params/doors/left/property"/>
        <factor>110</factor>
        <center>
            <x-m>-0.08856</x-m>
            <y-m>-0.24634</y-m>
            <z-m>0.55138</z-m>
        </center>
        <axis>
            <x>0.0</x>
            <y>1.0</y>
            <z>0.0</z>
        </axis>
    </animation>`

And somewhere, there is this code.

`

    <animation>
        <type>pick</type>
        <object-name>doorhandleext_left</object-name>
        <object-name>doorhandleint_left</object-name>
        <visible>true</visible>
        <action>
            <button>0</button>
            <repeatable>false</repeatable>
            <binding>
                <command>nasal</command>
                <script>c172p.leftDoor.toggle();</script>
            </binding>
        </action>
        <hovered>
            <binding>
                <command>set-tooltip</command>
                <tooltip-id>door-handle-l</tooltip-id>
                <label>Left Door Open/Close</label>
            </binding>
        </hovered>
    </animation>
`

I understand that the second block is the model i can use, but now, defining "sunshade_left" as the "pick" object to make it clickable.

But i don't understand the "property-alias" on the first block, nor the "nasal" on the second.

**Comment by thevirtualfer** [comment]
Gil.
I have a bunch o photos from @tigert. The problem is that our instruments are fixed in size. So, the proportions are all related to the instruments we have (assuming they are in the correct proportions). I merged all leftpanel instruments inside my working blender file and positioned they using this blueprint. The panel around is a bit small, in my opinion. But will model the panel and see.

**Comment by gilbertohasnofb** [comment]
> The problem is that our instruments are fixed in size. 

I don't understand why can't we resize them, but for sure you know this better than anyone else here. Let's see how when you finish modelling the panel as you say, I can't wait for it :smile:

**Comment by onox** [comment]
You should be able to scale objects.

@thevirtualfer The `alias` attribute tells FG to get its value from another property. So if you have:

```xml
<property>/path/to/property</property>
```

Then `/path/to/property` can contain a number, and that number will be used. But if you do:

```xml
<property alias="/params/something"/>
```

Then you can at the top of the file put:

```xml
<params>
    <something>/path/to/different/property</something>
</params>
```

In `Models/c172p.xml` it is used to keep all the `sim/multiplay/generic/` paths in one location (`<params>`) instead of having it in dozens of places. Imagine you're somewhere between line 4000 and 6000 and then you see this (suddenly):

```xml
<property>sim/multiplay/generic/int[10]</property>
```

Would you then immediately understand that it has to do with the pontoon damage? No :stuck_out_tongue_closed_eyes: 

Note that we're still using unnecessary `<property>` elements inside `<params>` and that we're using silly relative addressing (`../`) instead of absolute (`/`). (Which is why we often write `<property alias="../../params/something/property"/>`). The reason is just when I introduced this I didn't fully understand the technique yet.

**Comment by onox** [comment]
Btw, I have changed your code blocks to use ```xml, so that you get syntax highlighting.

**Comment by thevirtualfer** [comment]
> You should be able to scale objects.
But the instruments are proportional to real life instruments, aren't they ?
BTW, i will think for one moment about this "user exit".

> I have changed your code blocks to use ```xml

Hehehehe ! Thanks @onox. Learning a bit more.

> Then /path/to/property can contain a number, and that number will be used.

Almost understanding. But where i define the property ?  Lets suppose that the sunshade_left has two properties ( i.e. "raised" and "lowered" , or "0" and "1"). As far as i understand, i need to define this somewhere, to the animation block rotate it in positive or negative degrees. Am i wrong ? How much wrong ?

Thanks again.

**Comment by wlbragg** [comment]
Yes, we can define the property to be any name we want Look at c172p-set.xml line 693 and down, there we define several properties. Then we can use or alias that property.

**Comment by wlbragg** [comment]
So it might look like 
~~~xml
<sunshade>
    <left>
         <raised type="int">0</raised>
         <lowered type="int">1</lowered>
    </left>
     <right>
    ~~~
    </right>
</sunshade>
~~~

**Comment by thevirtualfer** [comment]
Veeeeeeeery thanks, @onox !!!!!

Will try tomorrow.

**Comment by thevirtualfer** [comment]
@wlbragg

Ok, it works.
Just to fix the idea : 
(1) Define the left_sunshade and right_sunshade properties in the c172p-set.xml

```xml
    <left-sunshade type="bool">false</left-sunshade>
    <right-sunshade type="bool">false</right-sunshade>
```

(2) define a pick animation (showing sunshade_left only, because sunshade_right is exactly same)

```xml
    <animation>
        <type>pick</type>
        <object-name>sunshade_left</object-name>
        <visible>true</visible>
        <action>
            <button>0</button>
            <repeatable>false</repeatable>
            <binding>
                <command>property-toggle</command>
                <property>left_sunshade</property>
            </binding>
        </action>
        <hovered>
            <binding>
                <command>set-tooltip</command>
                <tooltip-id>left_susnshade</tooltip-id>
                <label>Pilot sunshade %s</label>
                <mapping>on-off</mapping>
                <property>left_sunshade</property>
            </binding>
        </hovered>
    </animation>
```
Binding a simple property-toggle (first click makes it "true", second makes it "false", and so on")

(3) Finally, describe the desired animation based on the property value. If true, rotate it 110 degrees.

```xml
    <animation>
        <type>rotate</type>
        <object-name>sunshade_left</object-name>
        <property>left_sunshade</property>
        <factor>110</factor>
        <center>
            <x-m>-0.09642</x-m>
            <y-m>-0.24354</y-m>
            <z-m>0.5567</z-m>
        </center>
        <axis>
            <x>0.0</x>
            <y>1.0</y>
            <z>0.0</z>
        </axis>
    </animation>
```

Ok. Looking other controls, i guess the sound is given by binding a nasal script. Is it true ?

Very thanks, bro.

Still learning. If i done something wrong, please, tell me.

**Comment by wlbragg** [comment]
Looks good to me.
@thevirtualfer, feel free to enhance the skis I just pushed to the skis branch. I modeled those off of memory from a real aircraft equipped with them. The part that I would consider a good upgrade would be to replace the metal risers that attach to the axles. I have seen it done the way I modeled it, but I have also seen the use of rods instead of just a metal sheet. I think the rods would be an enhancement.
![rod](https://cloud.githubusercontent.com/assets/11069680/10061621/12ca73fa-6220-11e5-8d47-cb1642589390.jpeg)

**Comment by thevirtualfer** [comment]
@wlbragg.

I can see what to do, but it looks very good to me.

In fact, it pushed on me a very strong desire to stop modeling and fly for fun.

**Comment by thevirtualfer** [comment]
Guys.

Any problem if i incorporate the circuit breakers into the main model ?

It is because i am following this blueprint and repositioning buttons, etc.

![panel2](https://cloud.githubusercontent.com/assets/10703842/10115726/b50240d8-63e9-11e5-828a-68c70ee17f6d.png)

**Comment by thevirtualfer** [comment]
Still working. A LOT !


Cheers.

**Comment by onox** [comment]
> the breakers are wrongly spaced.

Do you mean just the distance between the circuit breakers, or also the ordering? I think the order should be OK.

When you're finished with repositioning everything, please indicate how much you have moved everything (instruments, circuit breakers, etc.) upward, left/right, and forward/backward. This is because I use the positions for the click sounds. It doesn't matter much if the old positions are used, but it would be more less precise.

**Comment by thevirtualfer** [comment]
Just the distance between then.

Is the fourth hole in this blueprint a lighter ?

I put the glovebox too.

You want the actual position or how much i moved ?

**Comment by onox** [comment]
> You want the actual position or how much i moved ?

Just the offset.

Is it correct you moved everything (instruments + yoke + circuit breakers/buttons/throttle/mixture) the same distance? In that case you can just give me the offset of the whole thing on the up/down (z?) axis. (1 number). I'll modify the `<position>` elements in `c172-sound.xml`.

**Comment by tigert** [comment]
Are you sure the blueprint is a c172? Or if it is, it might be the "hawk xp" variant which has a constant speed prop and 180 or was it 200hp? That would of course be yet another variant ;-) But I think that fourth hole is the (blue) propeller adjustment knob.

**Comment by tigert** [comment]
Actually it would be mixture which would be the rightmost, prop is between throttle and mixture usually.

**Comment by gilbertohasnofb** [comment]
@tigert 

> Actually it would be mixture which would be the rightmost, prop is between throttle and mixture usually.

But just to make it clear, you are still thinking that's not the blueprint of the model P but of another variant which has an extra lever, correct?

**Comment by tigert** [comment]
Usually the older cessnas just have empty space there. OH-CTL has a "fourth hole" because it has that weird flap lever there, not the usual "slot" lever on most 172's.

**Comment by tigert** [comment]
Gap: https://www.barnstormers.com/listing_images.php?id=968711&ZOOM=09be7d4b1eab29fedc095413adef4908

This is actually from the c172 panel thread in forum, this has constant speed prop and might be a 172. https://farm4.staticflickr.com/3059/2336149328_7e1e74b93b.jpg

**Comment by Juanvvc** [comment]
> Any problem if i incorporate the circuit breakers into the main model ?

In fact, I would prefer if the whole panel is removed from the main model for these reasons:

- Having all objects inside the blender file is very convenient for the modeller, but a lot less convenient for the people editing xml, nasal, textures and shaders.
- some collaborator willing to enhance just a small piece shouldn't open and edit the already very complex main .blend file.
- In we introduce variants in the future (constant propeller, glass cockpit), it is much easier to manage them if the panel is independent from the fuselage.

As a "middle-ground solution", I suggest exporting the main .blend file into different .ac files: (1) mainmodel.ac (including for example the fuselage and all the objects in the exterior) (2) interior.ac (including the seats, doors and things like these) and (3) panel.ac I believe these correspond roughly to the different layers in the current .blend file, so they can be easily exported separately.

**Comment by thevirtualfer** [comment]
@onox wrote : 
> Is it correct you moved everything (instruments + yoke + circuit breakers/buttons/throttle/mixture) the same distance?

No, it isn't.

The disposition we have are different from the blueprint. Buttons move an offset, mixture moves another offset, and so on.

@tigert wrote : 
> Are you sure the blueprint is a c172?

Yes. It is the 1981 blueprint.

> Or if it is, it might be the European Reims-manufactured "hawk xp" or "Reims Rocket" variant

Oh !
About this i don't know nothing.

Have you another blueprint ?

@ludomotico wrote :
> In fact, I would prefer if the whole panel is removed from the main model

Ok. No problems to me.

> Having all objects inside the blender file is very convenient for the modeller, but a lot less convenient for the people editing xml, nasal, textures and shaders.

I got. But i can say that not having all objects inside a blender file, is very convenient to people editing xml, nasal, textures and shaders. But a lot less convenient to the modeller. For who it must be convenient ?

There is another point here. Having a lot of independent textures, will cause more texture switches into the simulator graphic engine. Thorsten talk a bit about this in the forum. It affects a lot the frame rate. He gives an opposite guidelide. To merge all to less texture files, causing less texture switching.

What we do ?

What i am doing is : 

Modelling each peace (buttons, mixture lever, etc, etc) into a separated blend file, with the UV cut done and texturized.

Then, first i append the file into the main model.

Second, i copy the texture file to a main texture file, with the freedom to move to where it best fits, freedom to rotate it, etc, etc.

Third, i just move the UV cut to the proper place.

Each object has an individual blender file and an individual texture.

But, at the end, i got a single "main" texture file.

**Comment by thevirtualfer** [comment]
@tigert 

Ok, it is not a "hawk xp".
Here is the "hawk xp" blueprint.

![image](https://cloud.githubusercontent.com/assets/10703842/10269719/29d850ec-6ab5-11e5-83d0-c07a5e095209.png)

There is a lot of panel layout here in this site http://www.aircraftpanel.com/

As you are an experienced pilot, can you visit this site and give your opinion ?

No problem to move it all again. All i want is make it as equal the real life panel as i can.

**Comment by Juanvvc** [comment]
I think you shouldn't think about improving performance before checking if we have a problem with performance ;)

As far as I know, the .ac exporter script only exports the objects *currently visible*. Hence, you can have a single .blend file with all the objects. When you are satisfied with your work and want to export the model, then you select *only* layer 1 and export to exterior.ac. Then, you select *only* layers 2, 3 and 4 and export to interior.ac. Then, you select *only* layer 5 and export to panel.ac These layers are only an example, and of course this process *must be documented in a file such as README.blender*

This way, other modellers may open only the .ac file to check the object names or even do simple modifications. We still have to import these modifications into the .blend file, but I think it would be much easier than asking contributors to edit the whole .blend file to change a sigle button.

Please, this is only a proposal and I'm not even sure if it is a good idea! But I believe a new contributor will find the .blend file with all the interior, exterior and panel very scary in the beginning. Also, this way makes much easier for XML coders to check the names and positions of the different objects for the animations and shaders.

**Comment by thevirtualfer** [comment]
@ludo. I am not saying it will be difficult to do it. It will be so easy that i can do it in less than a entire hour.

Of course, we can define all separated and have only links on the main blend. In fact, it is the right way  to work.

The problem i am seeing is : 

As each object will have a single and possible small texture file, we will have a lot of individual texture that needs to be loaded individually. Thorsten said this is a expensive task (switching textures).

But ok. I consider it reasonable. But thinking with myself, i see that fsx has a small number of texture files. And X-Plane has a piece of pure art in this matter. The interior is really beautifully organized on a single texture file in a really smart and well done way. I recommend the modellers outside to just give a look on the X-Plane cut. Amazing. The plane itself is wrongly modeled. But the guys that do the cut and textures, are really good in what they do.

Of course we can try it.

One more point. If we encounter performance issues, will be difficult to avail if it is caused by the increase of the vert count or by the new way we done textures.

But i will continue modelling and doing some experimental cuts, just to see if textures fit on the place.

The retopo stage is almost finished. Just doing some minor changes (for example, the seats are very small, etc, etc).

Well. After all, separating the plane in small models, will affect ALL the textures anyway. A long talk with @gilbertohasnofb is coming.

Ah, Gil !

Do you have your working gimp files for the liveries and interior ?

It will help me a lot, helping you to convert the textures.

Cheers to all.

**Comment by thevirtualfer** [comment]
@ludo

Do you know if there is a "rotate" directive in wich we need not give the center ?

If not, do you know if it is very hard code one ?

Better saying, where the "rotate" directive is interpreted, at all ?

**Comment by thevirtualfer** [comment]
@ludo

As all you know, my english is not so good.

Rereading what you wrote, i am confuse for what you want.

(1) Each model or set of related models, will have an individual  .blend, a separated .ac, a separated .xml and a separated texture file. 

So, we will only include objects into our c172p.xml files, as each one will have your personal set of files to define himself.

OR 

(2) Each model or set of related models, will have a separated .blend file, not a .ac file and, when desired, a shared (or individual) texture file. So, we will not include the model into the xml file. We just link the objects into the c172p.blend file.

When someone needs modify a model, it can do it on the c172p.blend file or into the individual model .blend file. Both methods, will in fact, modify the individual model .blend file, as the linked objects are shared between c172p.blend and model .blend files.

To illustrate, i put the four panels (up_left, up_center, up_right an inf) into a separated panel.blend.

Then, i linked these objects into the main .blend file. These objects appears into the main .blend as blue, indicating they are links and are shared between files, as the picture shows. 

![image](https://cloud.githubusercontent.com/assets/10703842/10296059/95fbd60e-6b9b-11e5-8954-b1b4b8a0bff4.png)

Into the panel.blend, they appear as normal objects.

In this way, we will continue with a huge c172p.xml, some small .blend files (optionally, with corresponding .ac file) and a central c172p.blend formed basically with linked objects.

(3) A situation between (1) and (2).

Please, can you clarify this point ?

**Comment by gilbertohasnofb** [comment]
@thevirtualfer

> A long talk with @gilbertohasnofb is coming.

I am fearful but at the same time looking forward to the challenge :wink: 

> Do you have your working gimp files for the liveries and interior ?

Yes, we have a repository just for that (the xcf files are too large for the main repository): https://github.com/gilbertohasnofb/c172p-detailed-liveries
And by the way, you all have commit access to it.

**Comment by thevirtualfer** [comment]
Gil

Very good.

@ludo   ERRATA

Linked objects can be modified only into original .blend files.

**Comment by tigert** [comment]
Hmm, those panels at aircraftpanel.com are custom made aftermarket metal panels to replace the old panels that had the plastic "3d skin" on top that is often cracked, worn and dirty. While they likely use the same layout to simplify installation, I would not count 100% on the similarity.

OH-CTL for example has a different flap switch compared to the newer Cessnas. According to wikipedia, C172F (1965) introduced electric flaps that replaced the "car handbrake" flap lever on the floor, but the currently common "pre-select" flap lever was  only introduced in the C172N (1977) -  That flap lever is in the "fourth hole" slot on CTL's panel. On many other aircraft there is just an empty space.

But basically, to sum it up, how about having four defined "slots" for engine levers in the panel, that might change by variant:

  * Slot 1 is carb heat if the engine has a carburettor. Otherwise empty space. This is a also a bit smaller knob.
  * Slot 2 is throttle lever. Black and round.
  * Slot 3
    * Mixture in c172p
    * Propeller on FR172J Reims Rocket (210hp!)
    * Propeller on R172K Hawk XP (195hp)
    * Propeller also on the C172RG "Cutlass" (180hp **carburettor** + constant speed prop)
  * Slot 4
    * Mixture on constant speed prop aircraft
    * Empty slot on fixed pitch prop aircraft

https://en.wikipedia.org/wiki/Cessna_172 has quite a bit of information about the variants and powerplants and other changes between versions by the way. It would be fun to add the RG variant sometime once we don't have enough tasks in our hands ;-)

When you look at the differences between those c172 versions that are many, "I" means injection in engine variants: "O-320-D2J" has carburettor, "IO-360-L2A" is injected. If you want to know what the other codes mean, hold on to your hat and see https://en.wikipedia.org/wiki/Lycoming_O-320 :-)

**Comment by onox** [comment]
I wouldn't take any 'blueprints' you may find as the absolute, ultimate truth. They might be simplified or inaccurate. But that's my opinion...

**Comment by tigert** [comment]
Yeah, and each aircraft, especially one like a C172 that usually gets a lot of flight hours and years, does evolve as things get replaced, and upgraded, so there are differences here and there.

**Comment by thevirtualfer** [comment]
Ok.

But the blueprint, being not the ultimate truth, at least will give the general guideline of how things should be disposed.

It is difficult to me, without access to a real cockpit, figure out this.

Photos are really good stuff. The feeling is almost instantaneous got.

But this feeling may be different (i guess it is certainly different) to one that knows a the aircraft "live" from to another that knows it only by photos.

For example, these blueprints don't maintain the same distance between instrument holes, the fuel/oil holes don't match in size our current instruments, etc, etc.

If i follow the blueprint "AS IT IS", i will forced to scale instruments, etc.

But, it is still useful to me, to figure how things are disposed in a orthogonal way. Very difficult to get this from photos.

But i am tuned with this. Thanks to both of you.

**Comment by thevirtualfer** [comment]
I am in a congress. No time to model until october/30.

**Comment by gilbertohasnofb** [comment]
Sorry for the change of milestone above, it was an accident. Must get some sleep now...

**Comment by wlbragg** [comment]
>I am in a congress. No time to model until october/30.

Hey, where are your priorities? :laughing:

**Comment by gilbertohasnofb** [comment]
Hey all, I just got a message from @thevirtualfer and he asked me to share some news with you all: he told me that the new 3D model is basically done (though he still has to incorporate some of our recent modifications), but that he is having some serious problems with his internet and he can barely access anything (he can't use GitHub for instance). He says he plans to move to a new house in January and he hopes that the connection will improve, but apparently his neighbourhood in Brazil is quite problematic with internet connections. As soon as he has some news, he will either write directly here or try to send me some e-mail and then I once again repost it here, ok?

**Comment by wlbragg** [comment]
:+1:

**Comment by onox** [comment]
Is he able to otherwise just upload/email the `.blend` file (and textures?) and then let someone like @wlbragg check it and integrate it into the repository?

**Comment by gilbertohasnofb** [comment]
I sent him an e-mail asking this.

**Comment by gilbertohasnofb** [comment]
He wrote me next time he has some decent connection he will send me the model, and then I will create a branch for it and let you all know, ok?

**Comment by onox** [comment]
Yes, add it to a new branch, `bug-446` for example.

**Comment by gilbertohasnofb** [comment]
Okay, I will do that as soon as he sends me the model (don't know when).

**Comment by onox** [comment]
@gilbertohasnofb Does @thevirtualfer know about:

1. The oil cap
2. The side door on the right side of the cabin (for cabin air). See http://www.kentstreetshs.wa.edu.au/Images/General/cessna%20172SP.jpg
3. The elevator trim surface on the right elevator. See [1](https://commons.wikimedia.org/wiki/File:Labeled_Cessna_172_trim_tab.jpg) and [2](https://www.flickr.com/photos/pier-olivier_veilleux_cysg/9419510304) and [3](https://upload.wikimedia.org/wikipedia/commons/8/81/BF-Lento,_OH-CMO,_Cessna_172P_Skyhawk_II_%2815836462123%29.jpg)

?

**Comment by gilbertohasnofb** [comment]
I don't think so but I will write him an e-mail about it later today.

**Comment by gilbertohasnofb** [comment]
I wrote @thevirtualfer one more e-mail today, let's hope he would be able to send us his blend model soon enough. With each day, the modifications we will need to do on his model are accumulating -- given how much our repository moved in this last year -- but on the other hand his improvements looked so promising, particularly the cockpit. We have such a realistic aircraft and yet the main view has one of the ugliest 3D model problems we have (and which he fixed in his model): https://github.com/Juanvvc/c172p-detailed/issues/6

Fingers crossed then.

**Comment by wlbragg** [comment]
:+1:

**Comment by oMarcosLucas** [comment]
Is this task killed ?

**Comment by wkitty42** [comment]
On 05/08/2016 06:14 PM, oMarcosLucas wrote:
> Is this task killed ?

not AFAIK... this is all on thevirtualfer who's last post in this topic was Oct 
2015...

**Comment by gilbertohasnofb** [comment]
I got an e-mail from @thevirtualfer some days ago, and he is still having a lot of problems with his connection. Hopefully this will all be solvable soon...

**Comment by jarlarntzen** [comment]
I'm considering doing som modeling/retopo on c172, mostly the windscreen and sides and although I quite competent in Blender I have a few pretty basic issues:

1. Should I export all layers in the blend to .ac?
2. How can I properly test the exported .ac wrt appearance and functionality to ensure it is still working correctly?
3. Instruments are missing from my .ac when viewed in fgviewer. Is this due to missing layers (Q1) or error in export (Q2) or simply error in viewing procedure?

Thanks

**Comment by onox** [comment]
Glass is handled in #118. If you can fix the wing as well then I don't mind you take over the issue. You could split the front part of the wing (the leading edge) to be a separate object.

**Comment by onox** [comment]
Instruments are in separate files.

**Comment by onox** [comment]
> Should I export all layers in the blend to .ac?

You can choose the layers? Isn't it simply a matter of choosing the .ac format in the export menu?

**Comment by jarlarntzen** [comment]
Hi, again. What I'm really trying to ask is: 
What's a recommended way to load and test the c172p-detailed?
I've been struggling with loading the aircraft model using:
```
fgfs --prop:/sim/model/path='Aircraft/c172p-detailed/c172pa-set.xml'
platform default fg_root = Path "/usr/share/games/flightgear"
```
This just defaults it to the built-in c172.

Loading the .ac file works though and loads the right model into FGFS but this is not a true test since I'm not getting any instruments in the panel.
```
fgfs --prop:/sim/model/path='Aircraft/c172p-detailed/Models/c172p.ac'
platform default fg_root = Path "/usr/share/games/flightgear"
```
How do you go about loading and testing your creations?

**Comment by gilbertohasnofb** [comment]
Could it be that you are having a typo in the first command You wrote `c172pa-set.xml` but our set file is actually called `c172p-set.xml`. Maybe FG is then loading the default aircraft, which happens to also be the c172p.

> Loading the .ac file works though and loads the right model into FGFS but this is not a true test since I'm not getting any instruments in the panel.

I believe the best way would be to clone the repo, create a branch for your tests and modifications, then replace the .ac model in that branch, then finally load this aircraft in FG. If you use FGRun, you can simply add the directory to it by clicking first on the `Prev` button, then adding the directory of the c172p folder to FG_Aircraft. Then you should see two 172's in the aircraft load menu.

**Comment by jarlarntzen** [comment]
@gilbertohasnofb Thanks. I'll try your suggestions. BTW: my xml is actuall called `c172pa-set.xml`. I was just kinda hoping there would be a nice way to test the aircraft in CLI.
Is this how you all perform testing the aircraft?

**Comment by onox** [comment]
> fgfs --prop:/sim/model/path='Aircraft/c172p-detailed/c172pa-set.xml'

The aircraft folder should be named "c172p", not "c172p-detailed". You can start fgfs with `--fg-aircraft=path/to/a/folder/containing/all/aircraft`. That should override the default c172p. No need to mess with `--prop`.

You can also add all these commands to a file called `.fgfsrc` in your home folder.

**Comment by jarlarntzen** [comment]
@onox Thanks! I'll try that too. I think I'll document this on the wiki if it works well on my system.

**Comment by dany93** [comment]
No.
The aircraft folder name must be "c172p" (you have to rename your downloaded one I think).
The file (under the "c172p" folder) named (by default in our aircraft) "c172p-set.xml" serves to launch the aircraft from fgfs. Same as every aircraft from FG, the xxx-set.xml is the first read by FG.
In command line, that should be something like (under Linux, fgfs git version)
`./run_fgfs.sh --airport=KSFO --fg-aircraft=/path/to/your-aircraft-folder --aircraft=c172p`
which calls for c172p-set.xml.

**Comment by jarlarntzen** [comment]
@dany93 Thanks!
> The aircraft folder name must be "c172p" (you have to rename your downloaded one like @onox wrote it). This is because this path (starting with Aircraft/c172p) is often called for from files inside the aircraft folder.

This explains quite a lot. Spent 2 hrs testing with my folder dutifully named `c172p-detailed`. I really didn't expect the folder name to be the root cause for the missing instruments et al. Too bad. I'll test your suggestions later (possibly tomorrow). Thanks again 😄

**Comment by wkitty42** [comment]
On 08/15/2016 08:02 PM, Jarl Arntzen wrote:
> Hi, again. What I'm really trying to ask is:
> What's a recommended way to load and test the c172p-detailed?
> I've been struggling with loading the aircraft model using:
>
> |fgfs --prop:/sim/model/path='Aircraft/c172p-detailed/c172pa-set.xml' platform
> default fg_root = Path "/usr/share/games/flightgear" |
>
> This just defaults it to the built-in c172.

i'm late to this after losing the 5 volt line in our firewall's power supply 
yesterday... it is actually very simple to load other craft... just put them in 
your own custom craft directory and add that to the fg-aircraft path like so...

--fg-aircraft=/home/myuser/flightgear-dev/fgaddon;/home/myuser/myflightgear/Aircraft

then FGFS will look in fgdata as well as the given path(s)... i have both the 
default c172p and this development version and use either one with ease... 
depending on your launcher (i use the new --launcher), you should be able to 
hover over the craft in the list and see the path for it... that's how i tell 
the difference between these two ;)

**Comment by jarlarntzen** [comment]
1. @wkitty42 @onox @dany93 Thanks, all. This worked. --airport=KSFO --fg-aircraft=/path/to/Folder --aircraft=c172p .
Maybe I can do something with the dependency on folder name to be exactly "c172p". 
2. @wlbragg Thanks. I wont touch the object names at all and will carefully make sure any replaced objects keep their name. Currently I'm just looking at the opportunities for the months ahead. 
3. I tried appending the windscreen from [c182s](https://github.com/HHS81/c182s) but it is too wide and has the wrong shape. What blueprints were used to trace over for the c172p?
4. Is there a summary of Blender export instructions anywhere? What layers to export and so on?

**Comment by onox** [comment]
> Is there a summary of Blender export instructions anywhere? What layers to export and so on?

1. Make sure all objects that you want to export are visible (click the eye in the outline view)

2. Make sure paths of textures are relative, not absolute

3. File > Export > AC3D (.ac)

**Comment by onox** [comment]
Do you have a recent 2.7x version of Blender?

**Comment by jarlarntzen** [comment]
@onox I'm on Blender 2.76 at the moment. 
Regarding layers/objects to export: I suppose all layers/objects should be exported now, including the landing light cone etc.

**Comment by onox** [comment]
Yes, those cones are needed for Rembrandt AFAIK.

**Comment by jarlarntzen** [comment]
@onox Thanks. Nice to know. I just found it strange to see the BeaconOffX and strobe objects placed in the center of the cockpit but I'm just assuming they will be correctly placed at run-time.

**Comment by wlbragg** [comment]
>I'm just assuming they will be correctly placed at run-time.

Yes.
Also, it's been awhile since I have worked on that blend so I just checked it out to refresh my memory. There is only one object that is not being used, just leave it hidden. When you export, everything that is not hidden gets exported using File/Export/ac.

Another heads-up, when and if you start on redesigning the wings. We have a damage model on this aircraft that simulates damage to the wings and the landing rear. As far as the wings are concerned, you would need to also do the damage models. But we are planning on simplifying the wing damage. Right now there are to many duplicate wings with different damage modeled to them. There are so many because the wing is one piece. We want to divide the wings in to two halves in the future and lessen the amount of damaged wing duplicates. We just need to have a discussion when you get to the wings.

**Comment by onox** [comment]
Deduplicating the wings is handled in #592. Basically the wings should be split into an inner part (what you see when the outer parts are invisible after a big crash) and an outer part (one for the left and one for the right side). Per wing side, there is 1 for when the wing is OK, 1 for wing damaged, and 1 for wing collapsed. For the "broken" state, you simply hide all possible outer parts in the XML file so that only the inner part is visible.

**Comment by jarlarntzen** [comment]
Hi, all.
I've done a few things lately:

1.  **Comparing sources**
Collected and compared exterior diagrams and "blueprints" of C172 from the following:
 1. 1981 POH
 2. 1968-72 Service Manual 
 3. 1972 Service Manual
 4. Cessna 172 Skyhawk Blueprint at drawingdatabase.com
All blueprints are mostly consistent and overlap nicely apart from possible errors during scanning of the sources and some slight differences at the wheelbase.

2. **Comparing sources to current model**
    * Main wheels should be 20% larger than front wheel according to 1977 SM.
    * Entire model is rotated some 2.5 deg in Blender + base of main wheels are another 5 cm too high up which brings the  tail 40 cm closer to the ground than it should be. Tail bottom should be at approx 104 cm from ground but ends up at 64 cm from ground in-game.
    * Leading edge and wing is too narrow.
 * Wingspan is some 10 cm too wide.

I'd like to rotate the model to align with the SM and POH blueprints but this will also include the instrument panel  and updating the xml accordingly. Do anyone have experience with grouping multiple instruments and rotating them as a whole?

Remodeling the windscreen and wings is no problem but moving the wheels into the right position will entail rotating the plane to make room.

**Comment by gilbertohasnofb** [comment]
@jarlarntzen Thanks for the detailed work :+1:

>  Do anyone have experience with grouping multiple instruments and rotating them as a whole?

Maybe @wlbragg has the answer for that? Also this could possibly be related to https://github.com/Juanvvc/c172p-detailed/issues/6#issuecomment-239006965

**Comment by dany93** [comment]
Maybe I don't understand well the issue, but wouldn't be just a matter of setting the z-position in the FDM for the main gears and offsetting the associated animation to recover the wheel contact on the ground in the view?
_(it would be too simple...)_ Although I can't find again where the z-position is set vs the gear kit. :confused:

**Comment by HHS81** [comment]
@jarlarntzen 
Since I am the original author of the 3d-exterior model, I think I can explain some things:

> Main wheels should be 20% larger than front wheel according to 1977 SM.

Better take a look at the companies which produces tires for aircraft. They have listed the right sizes for each aircraft. If the aircraft will get a higher MTOW the size of tires will change. So there might be a difference between a 1977 and a 1981 Cessna 172. Goodyear has some good documents:
[https://www.goodyearaviation.com/search/tire-results-mfg.html?search=product&mfg=Cessna&model=172%20Skyhawk&size=&partnumber=&market=&returnformat=json](Cessna 172 Goodyear)
and
[https://www.goodyearaviation.com/resources/pdf/tire_specifications_7_2016.pdf](url)

> Entire model is rotated some 2.5 deg in Blender + base of main wheels are another 5 cm too high up which brings the tail 40 cm closer to the ground than it should be. Tail bottom should be at approx 104 cm from ground but ends up at 64 cm from ground in-game.

First I made the mistake to use the drawing as it was. It shows the aircraft at a specified weight sitting on the ground. 
I should have taken the image as the aircraft level in flight. I think this rotation was later correctd also by the model.xml.

The difference of the height of main wheels I think was due the handling of the compression animation. 
On the Cessna 182 S I have modelled the it according the drawing, looked how much the gear compression length actually in FGFS is, and added those to the contactpoints, and wrote the matching animation.

If I see all the problems related to the modelling I think it would be better to completly start from scratch. The more you fiddle with more .xmls and other stuff just to correct already wrong things the worser the perfomance load of the aircract will be. And it is already more worse than the Cessna 182 S.

**Comment by jarlarntzen** [comment]
Also, BTW. The reason for the difficulties with lining up the Skyhawk Blueprint is due to the projection axes not being level to the ground. The blue, vertical axis is at a straight angle with the horizontal axis and the top projection is looking straight down along with it, not straight towards the ground.

The same is the case with the parallel purple lines: Notice how, in the sideways view, the top line intersects with the bottom of the windscreen and goes straight through the middle of the elevator. 
The bottom line is at a tangent to the main wheels bases and intersects the front wheel at the bottom of the rim.

In Blender, when I align the top and front view to the blue axes in the side view, the aircraft body, windows etc appears mostly very nicely aligned with the blueprint apart from the bottom which may be some 2-3 cm too high up (and the wheels which don't align at all). Otherwise it's perfect.

![c172-skyhawk blueprint - axes 2](https://cloud.githubusercontent.com/assets/2108412/18182245/6d03d144-708f-11e6-9c99-c506839b7d8f.png)

**Comment by HHS81** [comment]
>  is due to the projection axes not being level to the ground.

That's common with modern blueprints. The frontview and the top view shows the aircraft in level, the sideview the aircraft on ground. Usually with MTOW

**Comment by wlbragg** [comment]
>Do anyone have experience with grouping multiple instruments and rotating them as a whole?

I might be able to help with this. We would need to talk about which instruments specifically and the details of the rotation/translations. While we are at something like this, we should take the opportunity to zero out any pitch, yaw and roll as to match the interior cockpit positions. We need to do this to allow for all the ALS lighting and shadow effects to work properly on these objects. We need to eliminate any xml rotations on all objects see https://github.com/Juanvvc/c172p-detailed/issues/790

**Comment by jarlarntzen** [comment]
BTW 2: [View all my sources for c172 blueprints and specs](https://www.dropbox.com/sh/s4t4fbu8uzywbpf/AABZcmk-jC8kmsRwXW98JmFqa?dl=0) In the SVG all blueprints are aligned to the 1977 Service Manual one.

@HHS81 Thanks for clarifying regarding the blueprint axes.
With regards to wheel size, I'm referring to page 8 of the c172 1977 service manual.
![cessna-172-service-manual-77-page_8](https://cloud.githubusercontent.com/assets/2108412/18228129/f9bb8b80-723e-11e6-81a2-31e30ded7ad8.png)
I'm quite sure this is referring to wheel rims of 6 inch diameter x 6 inch width for the main wheels and 5 inch diameter x 5 inch width for the nose wheel. According to the Goodyear table, this would yield an outside max diameter of 17.5 inch (44.45 cm) for the main wheels and 14.2  inch (36.68 cm) for the nose wheel.

@dany93 Yes. That will probably work just well, I'll give it a try. Thanks.

**Comment by dany93** [comment]
And it would give more room to pitch the aircraft up at landing, holding it close to stall. Currrently, touching the ground with the tail happens easily enough.

**Comment by onox** [comment]
> Entire model is rotated some 2.5 deg in Blender + base of main wheels are another 5 cm too high up which brings the tail 40 cm closer to the ground than it should be.

In `Models/c172p.xml` I see the following:

```xml
   7     <offsets>
   8         <pitch-deg>-3.0</pitch-deg>
   9         <z-m>-0.065</z-m>
  10     </offsets>
```

So if you rotate the model in Blender, you should modify/remove the `<offsets>` in the XML file as needed. Just mentioning it in case you didn't know about it.

> Do anyone have experience with grouping multiple instruments and rotating them as a whole?

If you are putting multiple instruments together in one `.blend` file, you could group textures together in order to minimize the number of textures. I noticed that there are some small instruments/objects that have multiple small textures. This requires a lot of calls to `glBindTexture()` which you want to avoid. It's better to have one 1024x1024 texture than four 512x512 textures.

**Comment by wlbragg** [comment]
@gilbertohasnofb 
This one appears to me a mix of issues, which we did fix a part of. However it looks like there are other concepts we didn't touch. I think I would leave this intact for now.

**Comment by gilbertohasnofb** [comment]
@wlbragg Sure we can leave it opened, but I personally dislike very much these huge issues touching too many problems. Perhaps we could consider closing this and opening new ones for whatever concrete things are still left to be done.
